src/main/java/org/springframework/data/couchbase/core/ReactiveReplaceByIdOperationSupport.java

```diff
@@ -81,19 +83,31 @@ public ReactiveReplaceByIdOperationSupport(final ReactiveCouchbaseTemplate templ
 		public Mono<T> one(T object) {
 			PseudoArgs<ReplaceOptions> pArgs = new PseudoArgs<>(template, scope, collection, options, domainType);
 			LOG.trace("replaceById {}", pArgs);
-			return Mono.just(object).flatMap(support::encodeEntity)
-					.flatMap(converted -> template.getCouchbaseClientFactory().withScope(pArgs.getScope())
-							.getCollection(pArgs.getCollection()).reactive()
-							.replace(converted.getId(), converted.export(),
-									buildReplaceOptions(pArgs.getOptions(), object, converted))
-							.flatMap(result -> support.applyUpdatedCas(object, converted, result.cas())))
-					.onErrorMap(throwable -> {
-						if (throwable instanceof RuntimeException) {
-							return template.potentiallyConvertRuntimeException((RuntimeException) throwable);
-						} else {
-							return throwable;
-						}
-					});
+			AttemptContextReactive ctx = template.getCtx();
+			TransactionResultMap map = template.getTxResultMap();
+			Mono<T> reactiveEntity;
+			if (ctx != null) {
+				reactiveEntity = support.encodeEntity(object)
+						.flatMap(converted -> ctx.replace(map.get(object), converted.getContent()
+						/*(GetOptions) pArgs.getOptions() */).flatMap(
```

Comment:
is this still todo? Are there any options on it? (if not maybe remove the comment?

Reply:
Yes - I still need to handle the options.